clients/js-legacy/src/actions/amountToUiAmount.ts

```diff
@@ -54,24 +59,51 @@ function calculateExponentForTimesAndRate(t1: number, t2: number, r: number) {
  * @throws An error if the sysvar clock cannot be fetched or parsed.
  */
 async function getSysvarClockTimestamp(connection: Connection): Promise<number> {
-    const info = await connection.getParsedAccountInfo(new PublicKey('SysvarC1ock11111111111111111111111111111111'));
-    if (!info) {
+    const info = await connection.getParsedAccountInfo(SYSVAR_CLOCK_PUBKEY);
+
+    if (!info?.value) {
         throw new Error('Failed to fetch sysvar clock');
     }
-    if (typeof info.value === 'object' && info.value && 'data' in info.value && 'parsed' in info.value.data) {
+
+    if (typeof info.value === 'object' && 'data' in info.value && 'parsed' in info.value.data) {
         return info.value.data.parsed.info.unixTimestamp;
     }
+
     throw new Error('Failed to parse sysvar clock');
 }
 
+/**
+ * Calculates the decimal factor for a given number of decimals
+ * @param decimals - Number of decimals
+ * @returns The decimal factor (e.g., 100 for 2 decimals)
+ */
+function getDecimalFactor(decimals: number): number {
+    return Math.pow(10, decimals);
+}
+
+/**
+ * Convert a UI amount to the atomic UI amount
+ * This is the amount of the mint in the smallest unit of the mint
```

Comment:
nit: Maybe just make it totally clear that this removes the decimal scaling only?